azure-mgmt-search/src/main/java/com/microsoft/azure/management/search/SearchService.java

```diff
@@ -0,0 +1,319 @@
+        extends GroupableResource.DefinitionStages.WithGroup<DefinitionStages.WithSku> {
+    }
+
+    /**
+     * The stage of the Search service definition allowing to specify the sku.
+     */
+    interface WithSku {
+      /**
+       * Specifies the sku of the Search service.
+       * @param skuName the sku
+       * @return the next stage of the definition
+       */
+      WithCreate withSku(SkuName skuName);
+
+      /**
+       * Specifies to use a free sku type for the Search service.
+       * @return the next stage of the definition
+       */
+      WithCreate withFreeSku();
+
+      /**
+       * Specifies to use a basic sku type for the Search service.
+       * @return the next stage of the definition
+       */
+      WithReplicasAndCreate withBasicSku();
+
+      /**
+       * Specifies to use a standard sku type for the Search service.
+       *
+       * @return the next stage of the definition
+       */
+      WithPartitionsAndCreate withStandardSku();
+    }
+
+    interface WithReplicasAndCreate extends WithCreate {
+      /**
+       * Specifies the sku of the Search service.
+       * @param replicaCount the number of replicas to be created
+       * @return the next stage of the definition
+       */
+     WithCreate withReplicas(int replicaCount);
+    }
+
+    interface WithPartitionsAndCreate extends WithReplicasAndCreate {
+      /**
+       * Specifies the sku of the Search service.
+       * @param partitionCount the number of partitions to be created
+       * @return the next stage of the definition
+       */
+      WithReplicasAndCreate withPartitions(int partitionCount);
+    }
+
+    /**
+     * The stage of the definition which contains all the minimum required inputs for
+     * the resource to be created (via {@link WithCreate#create()}), but also allows
+     * for any other optional settings to be specified.
+     */
+    interface WithCreate extends
+        Creatable<SearchService>,
+        Resource.DefinitionWithTags<WithCreate> {
+    }
+  }
+
+  /**
+   * The template for a Search service update operation, containing all the settings that can be modified.
+   */
+  interface Update extends
+      Appliable<SearchService>,
+      Resource.UpdateWithTags<Update>,
+      SearchService.UpdateStages.WithReplicas,
+      SearchService.UpdateStages.WithPartitions {
+  }
+
+  /**
+   * Grouping of all the Search service update stages.
+   */
+  interface UpdateStages {
+
+    /**
+     * The stage of the Search service update allowing to modify the number of replicas used.
+     */
+    interface WithReplicas {
+      /**
+        * Specifies the Replicas count of the Search service.
+        * @param replicaCount the replicas count; replicas distribute workloads across the service. You need 2 or more to support high availability (applies to Basic and Standard tiers only)
+        * @return the next stage of the definition
+       */
+      Update withReplicas(int replicaCount);
```

Comment:
The parameter type/name will infer that it's a count/number... I can refactor this but I though that it will look repetitive to capture "count" as part of the method's name.